nixos/modules/services/desktops/ayatana-indicators.nix

```diff
@@ -0,0 +1,54 @@
+{ config
+, pkgs
+, lib
+, ...
+}:
+
+let
+  cfg = config.services.ayatana-indicators;
+in
+{
+  options.services.ayatana-indicators = {
+    enable = lib.mkEnableOption (lib.mdDoc ''
+      Ayatana Indicators, a continuation of Canonical's Application Indicators
+    '');
+
+    packages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
```

Comment:
Is there common use cases where this would be more then one package and why isn't the default set to `ayatana-indicator-messages` or at lest the example

Reply:
There will be multiple indicator packages, both [upstream Ayatana ones](https://github.com/AyatanaIndicators/?q=ayatana-indicator-) and [Lomiri-specific ones](https://gitlab.com/ubports/development/core?filter=indicator) which provide different/further functionalities. As for the default, I think I'll need to rethink & retest abit how all of this can work inside & outside of Lomiri.

For more context:
- The upstream Ayatana ones have optional enhancements for Lomiri, which require building them and the `libayatana-common` they use. They use the same binary & service names as the "vanilla" versions, so having different variants of the same indicator packages in here would be :boom:.
- Outside of Lomiri, users should be able to chose which indicators they want.
- But when using Lomiri, some indicators are expected to be present to provide core functionalities (`ayatana-indicator-power` is its *only* way of providing a graphical logout & shutdown, AFAICT).

I'll try to ~~steal~~ *get inspired by* how Debian handles this. I assume they just enable the optional Lomiri enhancements and non-Lomiri use-cases deal with the additional deps. Which wouldn't be great, but so much easier to manage & wrap one's head around.

Reply:
- #262118 shall enable Lomiri features for regular `libayatana-common`, this should simplify things here.
- Users should still be able to choose which indicators they care about.
- Not all DEs support all indicators: In MATE I can get `session` to appear, but not `messaging` or `power`. In Lomiri, all 3 work fine.

I'll add an `example` and expand the `description` to explain these details. Populating the default with some indicators is something I don't see as too beneficial though.

Reply:
`example` added, `description` expanded.